### PR TITLE
8289003: JavaThread::check_is_terminated() implementation should rely on Thread-SMR

### PR DESCRIPTION
**Pull request by dcubed-ojdk**
A small fix to change `JavaThread::check_is_terminated()` to rely on Thread-SMR.
The change is textually trivial, but requires a bit of thinking about the JavaThread
lifecycle. Since this is a cleanup of some old Thread-SMR related changes, I also
took the time to cleanup some comments and white space while I was visiting the
uses of `_vm_exited`.

Mach5 Tier1 has no related failures. Mach5 Tier[23] testing has just started.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289003](https://bugs.openjdk.org/browse/JDK-8289003): JavaThread::check_is_terminated() implementation should rely on Thread-SMR

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [baf43a88](https://git.openjdk.org/jdk/pull/9502/files/baf43a887bdf5cca961ce7c570159c52d3e0e6f1)
 * [Patricio Chilano Mateo](https://openjdk.org/census#pchilanomate) (@pchilano - **Reviewer**) ⚠️ Review applies to [baf43a88](https://git.openjdk.org/jdk/pull/9502/files/baf43a887bdf5cca961ce7c570159c52d3e0e6f1)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9502/head:pull/9502` \
`$ git checkout pull/9502`

Update a local copy of the PR: \
`$ git checkout pull/9502` \
`$ git pull https://git.openjdk.org/jdk pull/9502/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9502`

View PR using the GUI difftool: \
`$ git pr show -t 9502`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9502.diff">https://git.openjdk.org/jdk/pull/9502.diff</a>

</details>

### CODE REVIEW

**Comment by dcubed-ojdk** [comment]
@dholmes-ora - Please review when you get the chance.

File: src/hotspot/share/runtime/javaThread.hpp [hunk 1]
```diff
@@ -567,12 +567,10 @@ class JavaThread: public Thread {
   bool is_exiting() const;
   // thread's GC barrier is NOT detached and thread is NOT terminated
   bool is_oop_safe() const;
-  // thread is terminated (no longer on the threads list); we compare
-  // against the three non-terminated values so that a freed JavaThread
-  // will also be considered terminated.
+  // thread is terminated (no longer on the threads list); the thread must
+  // protected by a ThreadsListHandle to avoid potential crashes.
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/javaThread.hpp:571]: // be protected ...

**Review by dholmes-ora — APPROVED** [review]
Looks fine.

Thanks,
David

**Comment by dcubed-ojdk** [comment]
@dholmes-ora - Thanks for the review and for catching the typo. Will fix shortly.
Got hit by a thunderstorm last night so I'm still trying to figure out what all got fried...

**Comment by dcubed-ojdk** [comment]
@coleenp, @fisk, @robehn or @pchilano - I need one more review from folks that have
been in the Thread-SMR code, if you can spare a few cycles...

**Review by pchilano — APPROVED** [review]
LGTM.

**Comment by dcubed-ojdk** [comment]
@pchilano - Thanks for the review.

**Comment by dcubed-ojdk** [comment]
Mach5 Tier[1-7] testing looks good.

/integrate

**Pull request closed without merging**